### ISSUE TITLE
Mismatch between layout/with-c-layout and clang's reported layout

### ISSUE THREAD

**Issue by phronmophobic** [IGJoshua/coffi#2]
I'm trying to use libclang to auto-generate struct layouts for use with coffi. As a sanity check, I wanted to make sure that the size of a struct reported by clang matched `mem/size-of`. It generally does, but I noticed there's a size difference when a struct includes pointers.

For example:
```c
#include <stdio.h>

typedef struct S1 {
    char c;
    void* p;
} S1;

int main(int n, char** args){
    printf("size: %lu\n", sizeof(S1));
}

```
The above gives a size of 16, but the same struct definition in coffi gives 9:
```clojure
(defalias ::S1
  (layout/with-c-layout
   [::mem/struct
    [[:c ::mem/char]
     [:p ::mem/pointer]]]))

(mem/size-of ::S1) ;; 9
```

It doesn't seem like there's anything wrong with the implementation. The byteAlignment of `ValueLayout/ADDRESS` is 1. I'm not really sure whether I'm using clang wrong or if there's something else going on. Either way, I thought I'd report it in case there is some improvement to be made.

**Comment by phronmophobic** [comment]
I'm also getting a mismatch for the following struct:

```c
typedef struct S2 {
    int i;
    void *data[2];
} S2;

int main(int n, char** args){
    printf("size: %lu\n", sizeof(S2));
}

```
This prints 24, but `coffi/size-of` reports 20.

```clojure
(defalias ::S2
  (layout/with-c-layout
   [::mem/struct
    [[:i ::mem/int]
     [:data [:coffi.mem/array [:coffi.mem/pointer :coffi.mem/void] 2]]]]))

(mem/size-of ::S2) ;; 20
```

**Comment by IGJoshua** [comment]
Ah, yeah, this was an issue I observed before that I've determined a fix for but it appears I applied it inconsistently. All the value layouts in the JVM-side default to an alignment of 1, which is incorrect, and I corrected this for primitive numbers, but it appears I forgot to do the same for pointers. I believe that the array in your second example should be fixed if I fix the alignment of pointers, but I'll check up on that as well.
